pkg/storkctlcli/storkctlcli.go

```diff
@@ -0,0 +1,161 @@
+)
+
+var dash *aetosutil.Dashboard
+
+const (
+	drPrefix = "automation-"
+	actionRetryTimeout = 10 * time.Minute
+	actionRetryInterval = 10 * time.Second
+)
+
+var (
+	migSchedNs = "kube-system"
+)
+
+func ScheduleStorkctlMigrationSched(schedName, clusterPair, namespace string, extraArgs map[string]string) error {
+	if namespace != "" {
+		migSchedNs = namespace 
+	}
+	cmdArgs := map[string]string{
+		"cluster-pair": clusterPair,
+		"namespace":   migSchedNs,
+	}
+	err := createMigrationScheduleCli(schedName, cmdArgs, extraArgs)
+	return err
+}
+
+func createMigrationScheduleCli(schedName string, cmdArgs map[string]string, extraArgs map[string]string) error {
+	factory := storkctl.NewFactory()
+	var outputBuffer bytes.Buffer
+	cmd := storkctl.NewCommand(factory, os.Stdin, &outputBuffer, os.Stderr)
+	migCmdArgs := []string{"create", "migrationschedule", schedName}
+	// add the custom args to the command
+	for key, value := range cmdArgs {
+		migCmdArgs = append(migCmdArgs, "--"+key)
+		if value != "" {
+			migCmdArgs = append(migCmdArgs, value)
+		}
+	}
+	if extraArgs != nil {
+		for key, value := range extraArgs {
+			migCmdArgs = append(migCmdArgs, "--"+key)
+			if value != "" {
+				migCmdArgs = append(migCmdArgs, value)
+			}
+		}
+	}
+	cmd.SetArgs(migCmdArgs)
+	// execute the command
+	logrus.Infof("The storkctl command being executed is %v", migCmdArgs)
+	if err := cmd.Execute(); err != nil {
+		if err != nil {
+			return fmt.Errorf("Error in executing create migration schedule command: %v", err)
+		}
+	}
+	// Get the captured output as a string
+	actualOutput := outputBuffer.String()
+	logrus.Infof("Actual output is: %s", actualOutput)
+	return nil
+}
+
+func PerformFailoverOrFailbackStlCli(action, namespace, migSchdRef string, skipSourceOp bool, extraArgs map[string]string) (error, string) {
+	var pfCmdArgs []string
+	if namespace != "" {
+		migSchedNs = namespace 
+	}
+
+	factory := storkctl.NewFactory()
+	var outputBuffer bytes.Buffer
+	cmd := storkctl.NewCommand(factory, os.Stdin, &outputBuffer, os.Stderr)
+	if action == "failover" {
+		if skipSourceOp {
+			pfCmdArgs = []string{"perform", "failover", "--migration-reference", migSchdRef, "--namespace", migSchedNs, "--skip-source-operations"}
+		} else {
+			pfCmdArgs = []string{"perform", "failover", "--migration-reference", migSchdRef, "--namespace", migSchedNs}
+		}
+	} else {
+		pfCmdArgs = []string{"perform", "failback", "--migration-reference", migSchdRef, "--namespace", migSchedNs}
+	}
+	if extraArgs != nil {
+		for key, value := range extraArgs {
+			pfCmdArgs = append(pfCmdArgs, "--"+key)
+			if value != "" {
+				pfCmdArgs = append(pfCmdArgs, value)
+			}
+		}
+	}
+	cmd.SetArgs(pfCmdArgs)
+	// execute the command
+	logrus.Infof("The storkctl command being executed is %v", pfCmdArgs)
+	if err := cmd.Execute(); err != nil {
+		if err != nil {
+			return fmt.Errorf("Error in executing perform %v command: %v", action, err), ""
+		}
+	}
+	// Get the captured output as a string
```

Comment:
Same as above, don't need lines from 112:114

Reply:
Here, we need output as we dont have a way to get name of action till now. Discussed this with omkar and he said as of now we have to get name using CLI parsing only as its a generated name.